### PR TITLE
24 refactor parse variable

### LINKED ISSUE #24: Refactor parseVariable from internal/shader/shader.go
Initial CCN is 30. This should be reduced to max 19 for the P+.

### PR DESCRIPTION
**Pull request by axellervik**
Split two large chunks of parseVariable into separate functions, decreasing CCN from 30 to 15 (a 50% decrease).

The CCN for the new functions, parseSingleVariable and parseMultipleVariable, have a CCN of 10 and 11 respectively. The reason the summed CCN of the 3 functions is increased is due to extra error checks when returning. 

Closes issue #24.

### CODE REVIEW

File: internal/shader/shader.go [hunk 1]
```diff
@@ -466,87 +541,25 @@ func (s *compileState) parseVariable(block *block, fname string, vs *ast.ValueSp
...
 		case len(vs.Values) == 0:
 			// No initialization
-
 		case len(vs.Names) == len(vs.Values):
 			// Single-value context
-
-			init := vs.Values[i]
-
-			es, rts, ss, ok := s.parseExpr(block, fname, init, true)
+			inits = append(inits)
```
Comment (aversey) [inline internal/shader/shader.go]: What is this for?

Comment (axellervik) [reply internal/shader/shader.go]: Placeholder before implementing function, mistakenly left there!

**Review by aversey — APPROVED** [review]

File: internal/shader/shader.go [hunk 2]
```diff
@@ -500,24 +497,24 @@ func (s *compileState) parseMultipleVariable(stmts *[]shaderir.Stmt, i int, t *s
...
 
-	if t.Main == shaderir.None && len(inittypes) > 0 {
-		*t = inittypes[i]
+	if t.Main == shaderir.None && len(*inittypes) > 0 {
+		*t = (*inittypes)[i]
 	}
 
-	if len(inittypes) > 0 && len(initexprs) > 0 && !canAssign(t, &inittypes[i], initexprs[i].Const) {
-		s.addError(vs.Pos(), fmt.Sprintf("cannot use type %s as type %s in variable declaration", inittypes[i].String(), t.String()))
+	if !canAssign(t, &(*inittypes)[i], (*initexprs)[i].Const) {
```
Comment (aversey) [inline internal/shader/shader.go:513]: I think this changes the behavior. Are you sure inittypes and initexprs cannot be empty here?

Comment (axellervik) [reply internal/shader/shader.go:513]: The len check was a band-aid fix by me to solve another thing I hadn't implemented correctly. Your comment about the variables being outside the for-loop made me realize my mistake and now inttypes and intexprs can't be empty anymore. I haven't taken the time to prove it formally, but `case len(vs.Values) == 0:`@ 548 should guarantee it. 

Tests work properly and that's what matters (;

**Pull request merged**